src/main/PayStackApi.cs

```diff
@@ -117,12 +117,13 @@ public TR Get<TR, T>(string relativeUrl, T request)
             var preparable = request as IPreparable;
 
             var queryString = "";
-            if (request != null)
-                queryString = "?" + request.ToQueryString();
-
+            
             if (preparable != null)
                 preparable.Prepare();
-
+            
+            if (request != null)
+                queryString = $"?{preparable.ToQueryString()}";
```

Comment:
Thanks for the catch! You're very right, the request's query, `request.ToQueryString()`, has to be generated after the request's preparation, if it's `IPreparable` (`preparable.Prepare()`).

However, I think this line should be: 
```c#
queryString = ...request.ToQueryString()...;
```